src/commands/registerCommands.ts

```diff
@@ -71,86 +72,101 @@ import { inspectVolume } from "./volumes/inspectVolume";
 import { pruneVolumes } from "./volumes/pruneVolumes";
 import { removeVolume } from "./volumes/removeVolume";
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function registerCommandWithActivity(commandId: string, callback: (context: IActionContext, ...args: any[]) => any, debounce?: number): void {
+    registerCommand(
+        commandId,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        async (context, ...args: any[]) => {
+            // eslint-disable-next-line no-unused-expressions, @typescript-eslint/no-floating-promises
+            ext.ams?.recordActivity('overallnoedit');
+
+            return callback(context, ...args);
+        },
+        debounce
+    );
+}
+
 export function registerCommands(): void {
     registerWorkspaceCommand('vscode-docker.api.configure', configureApi);
     registerWorkspaceCommand('vscode-docker.compose.down', composeDown);
     registerWorkspaceCommand('vscode-docker.compose.restart', composeRestart);
     registerWorkspaceCommand('vscode-docker.compose.up', composeUp);
     registerWorkspaceCommand('vscode-docker.configure', configure);
     registerWorkspaceCommand('vscode-docker.configureCompose', configureCompose);
-    registerCommand('vscode-docker.pruneSystem', pruneSystem);
+    registerCommandWithActivity('vscode-docker.pruneSystem', pruneSystem);
```

Comment:
I'm still considering using `TelemetryPublisher` along with some sort of categorization, instead of this. `registerCommand` would have to set that categorization to `'command'`, up in `vscode-azureextensionui` (but we'd be blocked on that currently due to #1925).

This way, `TelemetryPublisher` would be the entry point for `recordActivity`. Each of our additional categories, if they ever turn up, would set that categorization.

What do you guys think?

Reply:
It would also be blocked on `vscode-azureappservice` adopting version 0.30.0 of `vscode-azureextensionui`.

Reply:
@bwateratmsft not sure if I understand the alternative... what would `TelemetryPublisher` report?

Reply:
`TelemetryPublisher` could be the single entrypoint where `recordActivity` would be called, but we would need to know what kind of activity it is--so need something set on the telemetry properties.

Reply:
If `ITelemetryReporter` does end up being removed, `TelemetryPublisher` will no longer be possible.